lib/questions.rb

```diff
@@ -1,226 +1,292 @@
+  # e.g. 'bob'. So in the array ['bob', 'radar', 'eat'], there
+  # are 2 palindromes (bob and radar), so the method should return 2
+  def number_of_elements_that_are_palindromes(array)
+    array.delete_if {|word| word != word.reverse}
+    array.length
+  end
+
+  # return the shortest word in an array
+  def shortest_word_in_array(array)
+    array.min{|a,b| a.size <=> b.size }
+  end
+
+  # return the shortest word in an array
+  def longest_word_in_array(array)
+    array.max { |a, b| a.length <=> b.length }
+  end
+
+  # add up all the numbers in an array, so [1, 3, 5, 6]
+  # returns 15
+  def total_of_array(array)
+    array.inject {|sum, elem| sum + elem}
+  end
+
+  # turn an array into itself repeated twice. So [1, 2, 3]
+  # becomes [1, 2, 3, 1, 2, 3]
+  def double_array(array)
+    arr = array
+    arr.each {|num| array << num}
+  end
+
+  # convert a symbol into a string
+  def turn_symbol_into_string(symbol)
+    symbol.to_s
+  end
+
+  # get the average from an array, rounded to the nearest integer
+  # so [10, 15, 25] should return 17
+  def average_of_array(array)
+    total = 0
+    array.inject{|sum, elem| total = sum + elem}
+    average = (total / array.length.to_f).ceil
+  end
+
+  # get all the elements in an array, up until the first element
+  # which is greater than five. e.g.
+  # [1, 3, 5, 4, 1, 2, 6, 2, 1, 3, 7]
+  # becomes [1, 3, 5, 4, 1, 2]
+  def get_elements_until_greater_than_five(array)
+  end
+
+  # turn an array (with an even number of elements) into a hash, by
+  # pairing up elements. e.g. ['a', 'b', 'c', 'd'] becomes
+  # {'a' => 'b', 'c' => 'd'}
+  def convert_array_to_a_hash(array)
+    hash = Hash[*array]
+  end
+
+  # get all the letters used in an array of words and return
+  # it as a array of letters, in alphabetical order
+  # . e.g. the array ['cat', 'dog', 'fish'] becomes
+  # ['a', 'c', 'd', 'f', 'g', 'h', 'i', 'o', 's', 't']
+  def get_all_letters_in_array_of_words(array)
+  end
+
+  # swap the keys and values in a hash. e.g.
+  # {'a' => 'b', 'c' => 'd'} becomes
+  # {'b' => 'a', 'd' => 'c'}
+  def swap_keys_and_values_in_a_hash(hash)
+    hash.invert
+  end
+
+  # in a hash where the keys and values are all numbers
+  # add all the keys and all the values together, e.g.
+  # {1 => 1, 2 => 2} becomes 6
+  def add_together_keys_and_values(hash)
+    array = hash.to_a.flatten!
+    array.inject {|sum, elem| sum + elem}
+  end
+
+  # take out all the capital letters from a string
+  # so 'Hello JohnDoe' becomes 'ello ohnoe'
+  def remove_capital_letters_from_string(string)
+    string.delete "HJD"
+  end
+
+  # round up a float up and convert it to an Integer,
+  # so 3.214 becomes 4
+  def round_up_number(float)
+    float.ceil
+  end
+
+  # round down a float up and convert it to an Integer,
+  # so 9.52 becomes 9
+  def round_down_number(float)
+    float.floor
+  end
+
+  # take a date and format it like dd/mm/yyyy, so Halloween 2013
+  # becomes 31/10/2013
+  def format_date_nicely(date)
```

Comment:
Unused method argument - `date`. If it's necessary, use `_` or `_date` as an argument name to indicate that it won't be used. You can also write as `format_date_nicely(*)` if you want the method to accept any arguments but don't care about them.